includes/database/queries/class-edd-db-column.php

```diff
@@ -0,0 +1,200 @@
+	 * @access public
+	 * @var string
+	 */
+	public $type = '';
+
+	/**
+	 * Sets up the order query, based on the query vars passed.
+	 *
+	 * @since 3.0.0
+	 * @access public
+	 *
+	 * @param string|array $query {
+	 *     Optional. Array or query string of order query parameters. Default empty.
+	 *
+	 *     @type string   $name        Name of database column
+	 *     @type string   $type        Type of database column
+	 *     @type integer  $length      Length of database column
+	 *     @type boolean  $unsigned    Is integer unsigned?
+	 *     @type boolean  $zerofill    Is integer filled with zeroes?
+	 *     @type boolean  $binary      Is data in a binary format?
+	 *     @type boolean  $allow_null  Is null an allowed value?
+	 *     @type mixed    $default     Typically empty/null, or date value
+	 *     @type string   $extra       auto_increment, etc...
+	 *     @type string   $encoding    Typically inherited from wpdb
+	 *     @type string   $collation   Typically inherited from wpdb
+	 *     @type string   $comment     Typically empty
+	 *     @type boolean  $primary     Is this the primary column?
+	 *     @type boolean  $searchable  Is this column searchable?
+	 *     @type boolean  $sortable    Is this column used in orderby?
+	 *     @type boolean  $in          Is __in supported?
+	 *     @type boolean  $not_in      Is __not_in supported?
+	 * }
+	 */
+	public function __construct( $args = array() ) {
+
+		// Parse arguments
+		$r = $this->parse_args( $args );
+
+		// Set object parameters
+		foreach ( $r as $key => $value ) {
+			$this->{$key} = $value;
+		}
+	}
+
+	/**
+	 * Parse column arguments
+	 *
+	 * @since 3.0.0
+	 * @access public
+	 * @param array $args
+	 * @return array
+	 */
+	public function parse_args( $args = array() ) {
+
+		// Parse arguments
+		$r = wp_parse_args( $args, array(
+			'name'       => '',
+			'type'       => '',
+			'length'     => '',
+			'unsigned'   => false,
+			'zerofill'   => false,
+			'binary'     => false,
+			'allow_null' => false,
+			'default'    => '',
+			'extra'      => '',
+			'encoding'   => $GLOBALS['wpdb']->charset,
+			'collation'  => $GLOBALS['wpdb']->collate,
+			'comment'    => '',
+			'primary'    => false,
+			'searchable' => false,
+			'sortable'   => false,
+			'date_query' => false,
+			'in'         => true,
+			'not_in'     => true
+		) );
+
+		// Sanitize values
+		$r['name'] = sanitize_key( $r['name'] );
+
+		// Smart types intelligently set all parameters
+		if ( $this->is_smart_type( $r['type'] ) ) {
+			$this->set_smart_type( $r['type'] );
+
+		// Check allowed types
+		} elseif ( in_array( strtolower( $r['type'] ), $this->get_allowed_types(), true ) ) {
+			$r['type'] = strtoupper( $r['type'] );
+		}
+
+		// Return array
+		return $r;
+	}
+
+	/**
+	 * Return array of allowed column types.
+	 *
+	 * @since 3.0.0
+	 * @access private
+	 * @return array
+	 */
+	private function get_allowed_types() {
+		return (array) apply_filters( 'get_allowed_types', array(
```

Comment:
If we keep this filter and the one below, we need to prefix them with `edd_<something>`.

Reply:
I think I'll probably remove them for EDD3, and work on them later if there's a business need. We can just use the appropriate MySQL column types until then, and worry about this bit of sanity checking once we identify if this is even a problem.

Reply:
Opted to remove this filter entirely.